src/components/views/DownloadModModal.vue

```diff
@@ -293,9 +295,12 @@ let assignId = 0;
                 }
             }, async (downloadedMods: ThunderstoreCombo[]) => {
                 ProfileModList.requestLock(async () => {
+                    const tsOutdatedMods = outdatedMods.filter(value => !value.getImplicitlyInstalled())
+                        .map(value => ModBridge.getThunderstoreModFromMod(value, tsMods))
+                        .filter(value => value !== undefined) as ThunderstoreMod[];
```

Comment:
Something like `.filter((value): value is ThunderstoreMod => value !== undefined);` might work here without explicit type casting. Dunno if it matters much.

Reply:
Reason for this is that `.filter` returns `ThunderstoreMod[] | undefined` and `!` doesn't appear to resolve the issue. `!` may only apply to nulls rather than undefined?

Reply:
At least on my IDE using the `value is ThunderstoreMod` type guard is enough to tell TS that this particular `filter` now only returns `ThunderstoreMod`s, making the type cast unnecessary.